### PR TITLE
[FLYTE-486] Support selecting IDP based on the query parameter

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
## Overview
* Adds IDP parameter which allows to selectively login to a particular IDP
* The IDP parameter is exposed by okta which allows to selectively do Identity verification using the IDP
* Flyteadmin if finds this parameter then forwards to okta to complete the auth flow
* Each IDP uses different idp parameter hence made it configurable

## Test Plan
Tested on internal dogfood tenant and verified without the google tab in the login screen we  can login using the okta apps in the dashboard page
Screenshot 2024-02-05 at 1 15 26 PM

## Rollout Plan (if applicable)
Rollout to cloud and serverless-gcp

## Jira Issue
https://unionai.atlassian.net/browse/<project-number>

## Checklist
* [ ] Added tests
* [ ] Ran a deploy dry run and shared the terraform plan
* [ ] Added logging and metrics
* [ ] Updated [dashboards](https://unionai.grafana.net/dashboards) and [alerts](https://unionai.grafana.net/alerting/list)
* [ ] Updated documentation

### CODE REVIEW

File: flyteadmin/auth/handlers.go [hunk 1]
```diff
@@ -154,6 +154,12 @@ func GetLoginHandler(ctx context.Context, authCtx interfaces.AuthenticationConte
 				logger.Errorf(ctx, "Was not able to create a redirect cookie")
 			}
 		}
+
+		// Add the IDPQueryParameter to the URL if it is present in the request
+		idpQueryParam := authCtx.Options().UserAuth.IDPQueryParameter
+		if len(idpQueryParam) > 0 && queryParams.Get(idpQueryParam) != "" {
+			url = fmt.Sprintf("%s&%s=%s", url, idpQueryParam, queryParams.Get(idpQueryParam))
```
Comment (katrogan) [inline flyteadmin/auth/handlers.go:161]: can we use the url library to encode this?

url.Query.Add() https://pkg.go.dev/net/url#example-URL.Query

**Review by katrogan — COMMENTED** [review]
looks good, just one nit

File: flyteadmin/auth/handlers.go (hunk 1 above)
Comment (pmahindrakar-oss) [reply flyteadmin/auth/handlers.go]: Done

File: flyteadmin/auth/handlers.go [hunk 2]
```diff
@@ -154,7 +155,24 @@ func GetLoginHandler(ctx context.Context, authCtx interfaces.AuthenticationConte
 				logger.Errorf(ctx, "Was not able to create a redirect cookie")
 			}
 		}
-		http.Redirect(writer, request, url, http.StatusTemporaryRedirect)
+
+		idpUrl, err := url.Parse(urlString)
+		if err != nil {
+			errString := fmt.Sprintf("failed to parse url %q: %v", urlString, err)
+			logger.Error(ctx, errString)
```
Comment (katrogan) [inline flyteadmin/auth/handlers.go:162]: nit: why not use logger.Errorf here is we only use errString in one place?

**Review by katrogan — APPROVED** [review]

**Review by katrogan — DISMISSED** [review]

**Review by katrogan — APPROVED** [review]

**Pull request merged**